src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copyState = { ...state };
+  const allPreviousVersionsOfState = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        copyState = {
+          ...copyState,
+          ...action.extraData,
+        };
+        break;
+
+      case 'removeProperties':
+        for (const trash of action.keysToRemove) {
+          delete copyState[trash];
+        }
```

Comment:
Move this piece of code to separate function outside `transformStateWithClones` as function `removeObjectProperties`